photos/main.go

```diff
@@ -168,10 +180,67 @@ func runDrop(c *cobra.Command, args []string) error {
 	return nil
 }
 
+func runSplit(c *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("argument required: <num splits>")
+	}
+	n, err := strconv.ParseUint(args[0], 10, 32)
+	if err != nil {
+		return fmt.Errorf("unable to parse argument <num splits>: %v", err)
+	}
+	numSplits := int(n)
+	log.Printf("splitting photos database into %d chunks", numSplits)
+
+	db, err := openDB(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() { _ = db.Close() }()
+
+	if err := initSchema(db); err != nil {
+		log.Fatal(err)
+	}
+	ctx.DB = db
+
+	log.Printf("splitting table \"users\"")
+	for count := 0; count < numSplits; {
+		// Use the userID generation logic.
+		userID := 1 + int(rand.ExpFloat64()/rate)
```

Comment:
yeah. I'm still poking at it. the photos example has other problems running against a growing cluster (or rather, the cluster has problems), so I can't even tell how effective the pre-splits are.